src/features/user/ManageProjects/components/SiteGeometryEditor.tsx

```diff
@@ -0,0 +1,252 @@
+}
+
+const defaultZoom = 1.4;
+
+export default function SiteGeometryEditor({
+  geoJson,
+  setGeoJson,
+  setErrorMessage,
+}: Props): ReactElement {
+  const tManageProjects = useTranslations('ManageProjects');
+  const reader = new FileReader();
+  const mapRef: MapRef = useRef<ExtendedMapLibreMap | null>(null);
+  const [viewport, setViewPort] = useState<ViewState>(DEFAULT_VIEW_STATE);
+  const [mapState, setMapState] = useState<MapState>(DEFAULT_MAP_STATE);
+  const [isSatelliteMode, setIsSatelliteMode] = useState(false);
+  const [coordinates, setCoordinates] = useState<number[][]>([]);
+  const [isDrawing, setIsDrawing] = useState(false);
+
+  // Handle click to add point
+  const handleClick = useCallback(
+    (e: MapMouseEvent) => {
+      setErrorMessage(null);
+      if (!isDrawing) return;
+      const lngLat = [e.lngLat.lng, e.lngLat.lat];
+      setCoordinates((prev) => [...prev, lngLat]);
+    },
+    [isDrawing]
+  );
+
+  // Finish drawing (double click closes polygon)
+  const handleDoubleClick = useCallback(() => {
+    if (coordinates.length < 4) {
+      setErrorMessage(tManageProjects('errors.polygon.minimumPoints'));
+      setCoordinates([]);
+      return;
+    }
+    const closed = [...coordinates, coordinates[0]];
+    const newFeature: Feature<Polygon | MultiPolygon, GeoJsonProperties> = {
+      type: 'Feature',
+      properties: {},
+      geometry: {
+        type: 'Polygon',
+        coordinates: [closed],
+      },
+    };
+
+    setGeoJson((prev: SitesGeoJSON | null) => {
+      if (!prev) {
+        return {
+          type: 'FeatureCollection',
+          features: [newFeature],
+        };
+      }
+      return {
+        ...prev,
+        features: [...prev.features, newFeature],
+      };
+    });
+
+    setCoordinates([]);
+  }, [coordinates]);
+
+  useEffect(() => {
+    async function loadMapStyle() {
+      const result = await getMapStyle('default');
+      if (result) {
+        setMapState((prev) => ({ ...prev, mapStyle: result }));
+      }
+    }
+    loadMapStyle();
+  }, []);
+
+  const handleViewStateChange = (newViewState: Partial<ViewState>) => {
+    setViewPort((prev) => ({
+      ...prev,
+      ...newViewState,
+    }));
+  };
+
+  // Handle zooming to the project site
+  const handleZoomToProjectSite = () => {
+    if (geoJson) {
+      if (!mapRef.current) return;
+      zoomInToProjectSite(mapRef, geoJson, 0, handleViewStateChange, 2600);
+    } else {
+      setViewPort({
+        ...viewport,
+        zoom: defaultZoom,
+      });
+    }
+  };
+
+  // UseEffect: Run zoom logic whenever geoJson or mapRef changes
+  useEffect(() => {
+    handleZoomToProjectSite();
+  }, [geoJson, mapRef.current]);
+
+  const onMove = useCallback(
+    (evt: ViewStateChangeEvent) => {
+      handleViewStateChange(evt.viewState);
+    },
+    [handleViewStateChange]
+  );
+
+  return (
+    <div className={`${styles.formFieldLarge} ${styles.mapboxContainer2}`}>
```

Comment:
Since we're no longer using mapbox here, update the class name `mapboxContainer2`